src/main/java/org/mdcfg/provider/MdcProvider.java

```diff
@@ -522,6 +522,41 @@ public <T> Optional<T> getValueOptional(MdcContext context, String key, Function
         }
     }
 
+    /**
+     * Read property value and convert it to {@code Optional} of provided primitive class.
+     *
+     * @param context reading context {@link MdcContext}.
+     * @param key property name.
+     * @param clas Class in which result suppose to be converted.
+     * @return property value or null.
+     * @param <T> type in which value suppose to be converted
+     * @throws MdcException in case property not found or provided class is not primitive.
+     */
```

Comment:
there are 8 primitives in java: byte, short, int, long, float, double, char, boolean (Integer, Double, etc. are not). 
I don't think this method is even needed, as it not extensible (as all primitives and their autoboxed classes are defined already).  So I would prefer new method per each type for readablity and runtime type safety.

I think we could have such generic method for any class, but not type bounded to range of types.